bokehjs/src/coffee/models/axes/axis.coffee

```diff
@@ -283,14 +283,20 @@ class AxisView extends Renderer.View
     @visuals.axis_label_text.set_value(ctx)
     _apply_location_heuristics(ctx, side, orient)
 
+    x = sx+nx*standoff+nx*xoff
+    y = sy+ny*standoff+ny*yoff
+
+    if isNaN(x) or isNaN(y)
+      return
+
     if angle
-      ctx.translate(sx+nx*standoff+nx*xoff, sy+ny*standoff+ny*yoff)
+      ctx.translate(x, y)
       ctx.rotate(angle)
       ctx.fillText(label, 0, 0)
       ctx.rotate(-angle)
-      ctx.translate(-sx-nx*standoff+nx*xoff, -sy-ny*standoff+ny*yoff)
```

Comment:
It used to be prudent to avoid setting anything on the CTX necessarily, I'd have to go dig up the benchmarks I found, is it still the case, I don't know.